ui/src/api.ts

```diff
@@ -41,39 +44,84 @@ async function setupAPI() {
 
 const api = {
   async scry<T>(params: Scry) {
-    if (!client) {
-      await setupAPI();
-    }
+    try {
+      if (!client) {
+        await setupAPI();
+      }
 
-    return client.scry<T>(params);
+      const clientScry = await client.scry<T>(params);
+      useLocalState.setState({ subscription: 'connected' });
+      useLocalState.setState({ errorCount: 0 });
+      return clientScry;
+    } catch (e) {
+      useLocalState.setState({ subscription: 'disconnected' });
+      useLocalState.setState({ errorCount: errorCount + 1 });
+      throw e;
+    }
```

Comment:
Revert, scries aren't passed through the same channel system so their errors don't have bearing on the connection status